### ISSUE TITLE
Report Request - All Sites

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1437]
In GitLab by @kdjstudios on Apr 10, 2019, 12:41

**Submitted by:** Gary Pudles <lgpudles@aol.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/7975653

**Server:** Internal

**Client/Site:** All

**Account:** NA

**Issue:** 

I need to be able to run the Accounts Receivable Report and the Accounts Receivable As Invoice for the entire company.  It should be totals by company or detail sorted by site.

It should be limited to only those people with access to all sites.  Otherwise a person should only be able to run this for the site(s) they have access to.

Priority 1.5

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 10, 2019, 12:46

@smasih @amishra @pchaudhary @tim.traylor 

This was brought up briefly during the SD demo yesterday. Basically, we are needing to have the ability with just one click select all sites for the report. Initially this just needs to be done on the both the AR reports, but we will most likely need this on other reports too. 

@tim.traylor had mentioned that internally he believes there is a certain LDAP group code we can use to help know which users have access to all sites, but externally might take more development. 

Marking this for an estimate, but please bring up for discussion with @tim.traylor in the morning meetings if any further clarification is needed.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 10, 2019, 13:26

The LDAP group is '000'.  It is already being used in SAB to determine if a user has access to all sites.  Thx

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 12, 2019, 08:41

@kdjstudios I spoke with Rich and he realized that he forgot to request some of the features that go along with running all sites:
They need the ability to have all of the most recent billing cycles for selected sites included in a single report.  On the Accounts Receivables this is already happening, but it will need to be added to the Accounts Receivable as Invoiced.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 12, 2019, 09:16

I wrote:

> We just wanted to follow up. Tim was able to review this further with Rich and we have the following updates.
> 
> In regards to the Account Receivables report; adding the ability to select all sites we believe is straight forward and the support team is currently estimating the work. 
> 
> In regards to the Accounts Receivables as Invoices report; we ran into a few concerns on how a user would be able to select which billing cycles to run the report for as each site could have multiple cycles and they all could be on different dates. From the review with Rich we will just want this to run for the last billing cycle for each site and to have that cycle listed in the header of the report to further identify and distinguish. 
> 
> Please let us know if there are any concerns or questions. 

@smasih @amishra @pchaudhary 

Since the reporting data itself is not changing, just the report parameters, I did not create a mock up as I would believe it to be straight forward. However, if one is needed please specify your concerns to ensure I can address them in a mock up.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 15, 2019, 08:10

Gary wrote:

> I think Rich's approach is not correct.  We need to do this for ALL billing Cycles in a range not to exceed a calendar month


Rich wrote:

> For all the other AR reports there is no issue. 
> 
> The AR as invoiced is based on each site billing cycle.
> 
> So I don’t see how my approach is not correct.

Gary wrote:

> All sites are capable of having multiple billing cycles with different dates and different A/R.  In order to get a complete picture of the A/R as of the last billing for each site and each site cycle you need to run all cycles at each site.  Failure to do that will result in under reporting.  If it is easier you can subtotal by billing cycle I guess

Rich wrote:

> I agree.
> 
>  
> 
> What I am saying is when you run the AR as invoiced report for all sites at once, the report needs to capture the last billing cycle for all cycles for all sites

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Apr 29, 2019, 06:51

Hi [Kyle] ( @kdjstudios ) & [Tim] (@tim.traylor ),  

The full requirement of the ticket is still not clear to us.

* Accounts Receivables - Understood
  * We need to add a select option of all sites which shows all of the user access site's data together with the sorting by site's name.

* Accounts Receivables as Invoiced - Not Clear  
  * Is this report have to run for the last billing cycle for each site and to have that cycle listed in the header of the report to further identify and distinguish?

* We are mentioning a rough range for the estimation as per the current understanding of the ticket. 
  * `8-10 Hours`

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 29, 2019, 07:38

@pchaudhary Accounts Receivables as Invoiced - Not Clear:
Hi Piyush, as Rich stated "when you run the AR as invoiced report for all sites at once, the report needs to capture the last billing cycle for all cycles for all sites".  

So for each site selected it will include the lost recent of each of that site's cycles.  So if a site has 3 billing cycles then the last cycle for each of those 3 is included in the report.

This is the only way to see the AR (as invoiced) for every single customer regardless of the cycle they are part of.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 29, 2019, 10:40

@pchaudhary @smasih Hi Piyush and Sumeet - please make this the top priority and get it completed ASAP.  If it is done before our 11:00 meeting that would be preferable.  Thx, Tim

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Apr 30, 2019, 08:44

Hi [Tim] ( @tim.traylor ),  

We have completed a report Account Receivables on the Merge server. Still working on the second report, will let you know once it's done.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 30, 2019, 09:29

@smasih @amishra @pchaudhary 

This was reviewed today: Please provide updated estimates to accompany each item. 

* Update to also include that the report needs always be subtotaled by site (group by site with totals)

* Filter/sort option to subtotal by billing cycle
 
* Sort by account number or name
 
* Sort by largest and/or longest outstanding.

EDIT: URGENT Priority

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on May 1, 2019, 05:54

@kdjstudios @smasih As per our discussion this AM please complete this for tomorrow.  Also, Kyle, can you please test what is on the merge server as that is the completed original request. This way if you find anything we won't lose another day before it can be addressed.  Thx

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 1, 2019, 09:12

@amishra @smasih @pchaudhary @tim.traylor 

I was able to test both reports. And it does not appear we have estimate yet for any of the latest requests: http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1440#note_39287; so I presume none of those have been developed either. 

### Accounts Receivables
- All sites is an option in the drop down.
 - Appears to run for all sites data; unclear on how I could confirm this without having the run the report for every site and then manually merge together.
- Took about 1 minute to run.
 - Had about 5,600 results. Will we need to also do the limit to 4000 results then force download for this report too? Will this cause chrome to crash.


### Accounts Receivables as invoiced
- Once I selected "All Sites" it automatically ran. I presume for the last billing cycles.
- Billing cycle drop down was able to select, but nothing was in the drop down. Should we disable or hide this?
- Took about 1 minute to run.
 - Had about 5,600 results. Will we need to also do the limit to 4000 results then force download for this report too? Will this cause chrome to crash.
- Possible Header improvement: Why are the "Phone" and "Base Rate" labels not in their own column. This is how the production report is, but should they not be in separate columns?

![image](/uploads/7452e1d004ddfc106488f367718f2d67/image.png) 


In summary, I think this should be good for Rich to beta test.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 1, 2019, 09:38

I wrote:

> The support team has finished the original development to add the "All Sites" option in the drop down. They are still working on the requests from yesterday's meeting with Gary; but we would appreciate if you could do some testing with what is done so far to ensure everything with the all sites feature is working to your expectations. We currently have this setup on a development server, please use your same login information and the URL below to access. Thank you and please let us know if you have any questions or concerns. 
> 
> https://merge.sabilling.com/reports

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 1, 2019, 14:15

Rich wrote:

> Since there are no site identification at all, I am not able to comment on this all report.

I wrote:

> Thank you for looking at the reports. Just to clarify your response, I believe Gary's request from yesterday will resolve your concern. Gary requested to have the accounts grouped by site with subtotals. Will this resolve your concern, or what would you need to be updated on the report?


Rich wrote:

> I need to see the final product reflecting Gary's requests.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 2, 2019, 09:03

Hi @kdjstudios  & @tim.traylor 

The two of the main items are completed for both of the reports.

* Update to also include that the report needs always be subtotaled by site (group by site with totals) - Completed

* Filter/sort option to subtotal by billing cycle - Have some doubts over this. (Discussion)

* Sort by account number or name - Completed

* Sort by largest and/or longest outstanding. - Not Done Yet


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 2, 2019, 09:18

I believe the following format would be used when both site and billing cycle is subtotaled.

**Site A**
- Account A Billing Cycle A
- Account B Billing Cycle A
 - **Subtotal Billing Cycle A**
- Account C Billing Cycle B
- Account D Billing Cycle B
 - **Subtotal Billing Cycle B**

**Subtotal Site A**

**Site B**
- Account A Billing Cycle A
- Account B Billing Cycle A
 - **Subtotal Billing Cycle A**
- Account C Billing Cycle B
- Account D Billing Cycle B
 - **Subtotal Billing Cycle B**

**Subtotal Site B**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 2, 2019, 09:58

Sort by largest and/or longest outstanding 
- Four Choices: Account Name, Account Number, Largest outstanding; Longest outstanding.
- This is sort within the site and each of the billing cycle.
- This will be either the balance descending (Largest outstanding) or the age descending (longest outstanding).

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 2, 2019, 10:53

@amishra @smasih @pchaudhary @tim.traylor 

I was just able to test this and I see that we now have a site column. I am going to hold off on asking Rich to test again until we have more progress and subtotals. 

It was mention the filter option to sort by account name or number was completed, but I do not see it? I do see that it is by default sorting by site name, then account name. But as discussed today, we will want to allow the user to choose how to sort.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on May 3, 2019, 05:57

As discussed today, there is no billing cycle in play on the standard AR report, so that grouping/sub-total won't apply to that report.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 6, 2019, 05:45

Hi [Kyle] ( @kdjstudios ) & [Tim] ( @tim.traylor ), 

We have developed the requested sorting & view options for both of the reports.

We have also tested the calculation of the results by running an individual report for the site. Please check.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 6, 2019, 08:55

@pchaudhary @tim.traylor 

Thanks for the update here are my concerns:

## Accounts receivables as invoiced:

I had to wait to run the report then I could select the subtotals by billing cycle option and have to rerun the report again. Took almost 10 minutes to finally get the results. Why was this added as an option? It was not requested to be an option to my knowledge. 
- Also on the subtotal line for a billing cycle, we do not have the site name, just the billing cycle name; which we may want to have the site name, as most sites have identically named billing cycles. @tim.traylor what do you think?

Once I selected "All Sites" it automatically ran. I presume for the last billing cycle. Why is it automatically running? The user may want to adjust other options. 


## Both Reports:

I do not see the sort by balance or age option on either report. What is the status of this?

As previously mentioned, these reports do take a while to run and have over the 4000 results when I run it. Do we need to do the same force to download option on this report as we did on the invoice journal?

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on May 6, 2019, 09:26

@kdjstudios @pchaudhary Hi Kyle, Gary requested the new sub-totals and we all reviewed that on Thurs.

The report should not run until options have been selected and then the 'run report' (or whatever the button is) is clicked.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 7, 2019, 07:32

@kdjstudios @tim.traylor 

* We have changed the AR as invoiced report as requested to not run until options have been selected & then the GO button is clicked.  

* There is already a sort by balance option on AR as invoiced report & now we also added it for AR report.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 7, 2019, 09:49

Discussed today:

- Stop automatic running of the reports
 - This is completed. 

- Subtotal by billing cycle needs to be an option, should be already done.
 - Completed as an option
![image](/uploads/843a8515bc9b0c311657831edb7bbe82/image.png)

- Sort by Largest or Longest
 - This appears to be added to both reports. The only adjustment I would say is to use the same type of drop down for the AR.
![image](/uploads/e549a742b5a96b7312ab9971da74035c/image.png)
 - Also looks like the layout should be a bit more aligned for the AR as invoiced
![image](/uploads/1b1ac241467ed4982a10c6a83a5e6557/image.png)

- 4000 limit will be second stage
 - We will mark this ticket back for development once we release the other improvements.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 7, 2019, 09:52

I wrote:

> The support team has been able to complete the improvement requests. May you please look at the updated versions of the reports on our development server (link below) to confirm the functionality is working as you expected. Please let us know if you have any suggestions, questions, or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 8, 2019, 06:54

@kdjstudios 

> Sort by Largest or Longest
>
> - This appears to be added to both reports. The only adjustment I would say is to use the same type of drop down for the AR.

* Fixed



> Also looks like the layout should be a bit more aligned for the AR as invoiced

* Fixed

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 8, 2019, 08:06

Gary wrote:

> This report is slower than dirt.  How long should this take to run

> It actually is not showing.  Four minutes and nothing.

I wrote:

> Please keep in mind the report is now running for multiple sites (about 25 in total) which will extend the run time of the report by about 25x the original time as that is how many more sites it is now running for. From my testing yesterday (I also just retested right now) in regards to running the report for all sites it took about 2 minutes and had about 5,500 results. Yesterday when I had informed the support team and Tim about this concern it was discussed and determined that this is over the 4000 results Chrome is able to handle in a timely manor. They specified that we will have to do the same 'force to download' that we had to do with the invoice journal report when the result total exceeds the 4000 results that Chrome is capable of handling. However, we wanted to receive Rich's feedback on the report's content, layout, and filter options before proceeding on that new adjustment to ensure that is correct and we do not need any adjustments with the current improvements

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 8, 2019, 12:21

Rich wrote:

> Although it was slow, I was able to run the Accounts Receivable and download as CSV and it looks good.
> 
> When I tried to run the Accounts Receivable as invoiced, both checking off subtotal by billing cycle and not checking , the report came up on my screen, but when I tried to download as CSV, I get error messages twice. 
> 
> HTTP 500 error
> That’s odd... the website can’t display this page
> The site may be under maintenance or could have a programming error.
> Try this
> Go back to the last page
> Try contacting the website’s owner

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 9, 2019, 14:53

@amishra @smasih @pchaudhary @tim.traylor 

Rich wrote:

> I would like to add a few modifications.
> 
>  
> 
> Provide option to run reports without subtotal by site or billing cycle. In other words, I want the same information, just do not want any subtotals. I would like to see the site associated with each account though.
> Eliminate the following sites from the “All” selection.- Demo Site; Las Vegas Outbound; St Paul, MN
>  
> 
> Let me know if you have any questions about my request.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 9, 2019, 23:11

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 13, 2019, 24:08

Hi [Kyle] (@kdjstudios ), 


> Provide option to run reports without subtotal by site or billing cycle.

* We have added the functionality of without subtotals as requested by Rich. We requested you to please check & confirm if there's anything else to do.



> Eliminate the following sites from the “All” selection.- Demo Site; Las Vegas Outbound; St Paul, MN

* We have also eliminated the above sites

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 13, 2019, 10:28

@pchaudhary 

Thank you for the update. I was able to run the report and the two updates appear to be working as expected. I have reached out to Rich for his testing and approvals. 

I wrote:

> The support team was able to update the reports to allow for the subtotal by the site and/or billing cycle to be an checkbox option for the user to check. Also they have removed the requested sites from the report. Please test the reports out on the below development server and let us know if you have any feedback. 
> 
> https://merge.sabilling.com

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 14, 2019, 10:10

Rich wrote:

> So, only open item from me is to add the same functionality to the Aging report, which is used for collection purposes. I believe a request to add this report to the list of the AR reports modified was already been submitted by Tim.


@smasih @amishra @pchaudhary @tim.traylor 

Marking this for announcement, however, I do not believe we have a ticket opened to do this same functionality to the Ageing report too. If the team knows of the ticket, please link it here, if there is not one lets create a separate ticket.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 14, 2019, 23:12

@kdjstudios 

I'm not able to see any ticket regarding do this same functionality to the Aging report too. Please create a ticket.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 15, 2019, 10:30

@pchaudhary @amishra @smasih @tim.traylor 

Rich wrote:

> The reports still include the following sites that need to be removed.
> 
> Nashville, TN
> 
> Textgen

I believe we have an existing ticket that was discussing removing most if not all of these 'terminated' sites. Would it possibly be better to finish that ticket, then do custom removal of sites on each report? #540

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 15, 2019, 11:09

Gary wrote:

> Just to confirm, has all the triad billing been taken out of Nashville's database?

Rich wrote:

> The last billing cycle for Nashville was Feb 2018.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 16, 2019, 07:34

@kdjstudios 



> Rich wrote:
>
> > The reports still include the following sites that need to be removed.
> >
> > Nashville, TN
> >
> > Textgen
>
> I believe we have an existing ticket that was discussing removing most if not all of these 'terminated' sites. Would it possibly be better to finish that ticket, then do custom removal of sites on each report? #540

* For now, we have removed the above sites from these reports only, as we have to look into that code separately.


* We have also got an email from Rich on the Support to remove the **Birmingham, AL** also from those reports so we have removed that also.

* We request you to please check on the Merge server.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 16, 2019, 08:26

I wrote:


> May you please check the updated report with the sites excluded as requested. Thank you and we look forward to your response. This is available for your testing on the following development server.
> 
> https://merge.sabilling.com/

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 16, 2019, 16:22

Rich wrote:

> Looks good.
> 
> Can we get all these reports moved to production ?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 23, 2019, 23:32

@kdjstudios 

We have successfully released this for the Internal Production server only.
As discussed on the last Tech call (23rd May), that both of the production servers (Internal & External) are out of the sync at the moment due to the SD release on the Internal only, so we have to release this feature for only the Internal server.

Once we have a call to release the SD for the External clients, then we can release this for External server also.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 24, 2019, 08:14

I wrote:

> This has been released to production. May you please confirm everything is working as expected. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 24, 2019, 12:46

Rich wrote:

> An issue with the AR as invoiced report.
> 
> Allentown- mo master 4/21/19 ?
> 
> Atlanta-  There is a 5/26/16 included - This was an old montly cycle that is no longer used.
> 
> Las Vegas- need to exclude this site
> 
> Memphis- Need to exclude this site.


I wrote:

> Thank you for the update, i will provide this to the support team, However, I am concerned with why we would exclude the Memphis site? I believe the Memphis site is used for both Memphis and West Memphis billing now.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 24, 2019, 13:28

Rich wrote:

> That is not correct. What I meant to say is that we need to exclude the Memphis 7/28/18 billing cycle. I do not even see a 7/28/18 billing cycle in the system?

I wrote:

> I also do not see an 7/28 cycle for 2018 for the Memphis site. I will notify the support team and keep you posted with their updates.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 24, 2019, 13:29

@amishra @amishra @pchaudhary @tim.traylor 

Marked back for development. Please see the above comments.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 27, 2019, 23:41

Hi [Kyle] ( @kdjstudios ),   

We are totally in the loss.  We can't see anything wrong with the developed report.  

> An issue with the AR as Invoiced report.
>
> Allentown- mo master 4/21/19?

* What does this mean **mo master**. I guess it's **no** but we can clearly see the 4/21/19 Master for the Allentown Site.  
  * ![image](/uploads/fe56189a5994523e6cf7ebee6c317bf6/image.png)


> Atlanta-  There is a 5/26/16 included - This was an old monthly cycle that is no longer used.

* This thing is nothing to be related with the report. All of the billing cycles have to be shown on the select box column of the billing cycle.
* For Rich's request, we have to delete the above billing cycle which can cause further issues with the system flow.

* Need to discuss with [Tim] (@tim.traylor ).  



> Las Vegas- need to exclude this site

* There are 2 sites containing Las Vegas. 
  1.  Las Vegas Outbound.
  1.  Las Vegas, NV

* Which of them need to be removed from the report, please confirm.



> Memphis- Need to exclude this site.  
> What I meant to say is that we need to exclude the Memphis 7/28/18 billing cycle. I do not even see a 7/28/18 billing cycle in the system?

* There are 2 sites including Memphis in its name.  
  1.  Memphis, TN
  1.  West Memphis, AR

* Memphis, TN do not have any billing cycle with 07/28/2018 Master.
* **West Memphis, AR** do have a Master billing cycle with 07/28/2018 date.  
  * ![image](/uploads/d7f02d8db34049d0f623925f22e42d2c/image.png)


* We request to Kyle to get these answers from Rich so that we can close this one ASAP.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 28, 2019, 14:11

@pchaudhary @tim.traylor @amishra @smasih 

- Allentown: I see there is a stand alone billing cycle, but when I ran with "All Sites" it was not included, only the NCSM sub billing cycles are shown. [accounts_receivables_invoiced_05_28_2019.csv](/uploads/618299b7f24202b425a74a82c34298c7/accounts_receivables_invoiced_05_28_2019.csv)

- Atlanta: Please discuss with @tim.traylor on what we need to do. I do not believe we should be including billing cycles from three years ago.

- Las Vegas: Please remove all Las Vegas sites from the "All Sites" drop down. 

- Memphis: Please remember that the West Memphis site merged into Memphis site. We should not be displaying any West Memphis results in the "All Sites".

Please let me know if you have any questions.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 31, 2019, 05:32

@kdjstudios @tim.traylor 

> Allentown: I see there is a stand-alone billing cycle, but when I ran with "All Sites" it was not included, only the NCSM sub-billing cycles are shown.

* We would like to clarify that there it was mentioned multiple times on the ticket (http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1440#note_39158, http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1440#note_38376) that the `All Sites` option it captures the last billing cycle for all cycles for all of the sites.  
So, here the last **Master Billing Cycle** for the **Allentown** is **05/21/2019 Master** due to which **04/21/2019 Master** is not included there.


> Atlanta: Please discuss with @tim.traylor on what we need to do. I do not believe we should be including billing cycles from three years ago.

* Need to discuss.

> Las Vegas: Please remove all Las Vegas sites from the "All Sites" drop down.

* Excluded.

> Memphis: Please remember that the West Memphis site merged into Memphis site. We should not be displaying any West Memphis results in the "All Sites".

* Removed West Memphis from the **All Sites** option.



Thanks

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on May 31, 2019, 07:51

@pchaudhary @kdjstudios Kyle - I asked Piyush to add an 'active' check to including billing cycles in the 'All Sites' AR report.  For Atlanta, since that is an old/unused cycle they should be deactivating it, which will then exclude it from the report.

Kyle, can you pls explain to me the issue with Allentown and the 4/21 cycle?  I see the 5/21 cycle on the subtotaled report, and that was the last cycle, so I'm not clear on the problem.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 31, 2019, 08:30

@tim.traylor 

Thanks for the update on the active check, that makes perfect sense. 

In regards to the Allentown; that sites has three billing cycles: Master, NCSM 1st, and NCSM 15th. Only the NCSM cycles are on the report, the Master is not included, but it needs to be. Please see the exported file in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1440#note_41024 you can clearly see there is only the NCSM cycles.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 3, 2019, 04:04

@kdjstudios @tim.traylor 

> Thanks for the update on the active check, that makes perfect sense.

* We have developed, tested & released the active check functionality on the Internal Server. It's just a small change to add an active check.
  * We request to the client to deactivate the sub-billing cycle which they don't want to see.  

> In regards to the Allentown; that sites have three billing cycles: Master, NCSM 1st, and NCSM 15th. Only the NCSM cycles are on the report, the Master is not included, but it needs to be. Please see the exported file in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1440#note_41024 you can clearly see there are only the NCSM cycles.


* We dug into the code & found the reason of the discrepancy:
  * As the report name posted by Kyle, we can clearly see that the report was pulled on 28^th May 2019:  
      * There was a billing cycle 05/21/2019 Master with no invoices under that billing cycle that time.
      * All of the invoices under the 05/21/2019 Master Billing Cycle was created on the 29^th May.
      * So, the thing is that at the time of report pulled, there was a billing cycle 05/21/2019 Master with no invoices, so that's the reason we can't see the 05/21/2019 Master Billing cycle on the report.
      * However, if you can pull the report now, you can see easily that there is a 05/21/2019 Master Billing Cycle with lots of invoices under that. - [accounts_receivables_invoiced_06_03_2019.csv](/uploads/904fde0f0c5220816bb304106f1ac735/accounts_receivables_invoiced_06_03_2019.csv)

Hope this helps.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 3, 2019, 10:02

@pchaudhary @tim.traylor 

Seems we have found a bug. If we have a (new) billing cycle with no invoices what should we do? Display a note/line to on the report that the cycle has no invoices? Or display the billing cycle previous? Currently by not displaying anything is confusing Rich (and I) so I would believe we will need to do something.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 3, 2019, 23:29

@kdjstudios 

> Seems we have found a bug.  

* We don't think it should come under the bug. We had already discussed this so many times that the issue can be reported as a bug only if the running thing is not as proposed & this new billing cycle concern was not in any of us heads at the time of development & even while testing. 

Let's discuss this on Thursday call.

> If we have a (new) billing cycle with no invoices what should we do? Display a note/line to on the report that the cycle has no invoices?

* Yes, we can do this either any of the above-proposed things.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 4, 2019, 09:54

Rich wrote:

> Any update on this issue?

I wrote:

> The support team has the following updates for each of your concerns. We will keep you informed with all further updates. Please let us know if you have any questions or concerns.
> 
> Allentown - Missing Master Billing cycle: The support team found that the last created cycle did not have any invoices yet created, thus why no invoices were showing. We are currently discussing if in this scenario the previous billing cycle should display, until invoices are generated.
> 
> Atlanta: The report has been updated to only include active Sub Billing Cycles. If old billing cycles are still showing, the site will need to deactivate that billing cycle on the site edit page. 
> 
> Las Vegas: The support team has updated and this should be removed now. 
> 
> West Memphis: The support team has updated and this should be removed now.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 5, 2019, 10:03

@pchaudhary @tim.traylor 

That seems like a reasonable estimate to update this. Marking for approved.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 5, 2019, 08:04

Rich wrote:

> Regarding the Allentown scenario, are you saying that the billing cycle was created but not finalized yet?  I would think that until the billing cycle is finalized, the prior billing cycle will be considered the last one and be selected?

I wrote:

> Yes, that is exactly what is happening. We are currently reviewing what all it will take to make those additional changes to the report and we will follow up with the support teams progress.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 5, 2019, 08:06

@smasih @amishra @pchaudhary @tim.traylor 

It looks like Rich agrees that we should only be displaying the last/most recent finalized billing cycle. Do we need to discuss this on Thrusday anymore or can we mark this back for development?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 5, 2019, 08:08

@kdjstudios 

We need the final call from @tim.traylor to confirm. 

We can go with the development tomorrow if he approves.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 5, 2019, 08:21

|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|2|Concepts and Implementation Well Understood|2.5|Low Impact|0.5|3|

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 6, 2019, 05:09

@kdjstudios 

We have developed the functionality as requested to only display the last/most recent finalized billing cycle.

We have also uploaded the latest production DB on the merge2 server for the testing purpose. Please have a look so we can release this on the upcoming release.

* We also would like to ask that should we release the New All Sites changes to the External server also as we are going to release SD this week on production?

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 6, 2019, 06:22

@kdjstudios @tim.traylor 

Whatsapp Chat - 

**Tim** - Hi Guys - I understand Rich submitted a ticket about the sorting problem on the AR reports?  The fact that you can't sort by account anymore because you have the 'sort by AR' always apply.  Is that being fixed for the release as well?  It makes the report extremely difficult to use, so 'Yes' would be a great answer :)

**Sumeet** - AR report does not affect the overall system so Yes we can do a release as Soon as it's fixed and confirmed.

Please find the estimate to fix this. Although as it's urgent, we're moving towards the development.

|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|Medium-1|3|Concepts and Implementation Well Understood|4.5|Low Impact|0.5|5|


* There were 3 reports to change for that.



* The functionality we are going to develop will provide an ability to User to select either any of the Filter or both.

* If none of the filters will be selected then the default filter will be **Sort By AR - Largest** then **Account Sort By - Name**.

* If both filters will be selected then also - it works as it **Sort By AR** first & then **Account Sort By**.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 6, 2019, 07:32

@kdjstudios 

We have fixed the Account Receivables report on the Merge2 server itself for the New changes mentioned above.

Could you please check with the functionality & provide us feedback.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2019, 13:11

#1492 opened the new ticket for sorting issue. will mark it as secondary.

I tested the updated version, but I think we need to clarify how this will work with Rich/Gary first before making changes without a clear request. 

My thoughts:

1. How can there be two sorts? as only one can actually occur right? If we do one then the other, it really isn't doing both. its just doing the first one...?

1. Should all the sorts options simply be placed into one drop down?

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 6, 2019, 13:42

@kdjstudios @pchaudhary It was never supposed to be two sorts.  It was supposed to be one sort with four options. http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1440#note_39497

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 7, 2019, 03:55

@kdjstudios @tim.traylor 

We have fixed the Last Finalized Billing Cycle request, Open New Tab request & Sorting changes request on the Merge2 server for all of the 3 reports.

* Accounts Receivables
* Accounts Receivables as Invoiced
* Aging

* Please have a look on the Merge2 Server.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 7, 2019, 07:09

@kdjstudios 

We also request you to please confirm if we have to release these changes with tomorrow's release.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 7, 2019, 11:25

These all appear to be working as expected. Marking as ready.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 8, 2019, 04:38

@kdjstudios 

We have successfully released this on both of the production servers. Please check.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 8, 2019, 10:21

@pchaudhary 

Thank you. all three reports appear to be working on both servers.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 11, 2019, 12:37

I wrote:

> The support team has made the updated to the report to fix the sorting and also to ensure we are not displaying billing cycles that do not have invoices yet generated. May you please confirm all is working to your expectations. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 19, 2019, 15:09

closed
